src/attr/_make.py

```diff
@@ -1068,7 +1064,14 @@ def make_class(name, attrs, bases=(object,), **attributes_arguments):
     else:
         raise TypeError("attrs argument must be a dict or a list.")
 
-    return attributes(**attributes_arguments)(type(name, bases, cls_dict))
+    post_init = cls_dict.pop("__attrs_post_init__", None)
+    return attributes(
+        these=cls_dict, **attributes_arguments
+    )(type(
+        name,
+        bases,
+        {} if post_init is None else {"__attrs_post_init__": post_init}
```

Comment:
This is needed *at least* for our hypothesis strategy…

Reply:
Hm, this is technically a new feature?

So we should either document it, or if we don't want it, I can change our Hypothesis strategies.

Reply:
Good Question!  It seems useful?

Reply:
I think feature parity is good even though using it like this is kinda awkward. Just document that it should be a callable that takes one argument, self.

Reply:
I think we should make it a proper argument to make_class, agreed?  (I’m gonna land this as it is but we should make it nicer so I'm not documenting it)